src/index.ts

```diff
@@ -66,23 +105,34 @@ const convertAction = (
  * @returns This function returns a component object that has the shape the Prismatic API expects.
  */
 export const component = <T extends boolean>(
-  definition: Omit<Component<T>, "actions"> & {
-    actions: Record<
+  definition: Omit<Component<T>, "actions" | "triggers"> & {
+    actions?: Record<
       string,
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      ActionDefinition<any, boolean, PerformReturn<boolean, any>>
+      ActionDefinition<any, boolean, ActionPerformReturn<boolean, any>>
+    >;
+    triggers?: Record<
+      string,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      TriggerDefinition<any, boolean, TriggerResult<boolean>>
     >;
   }
 ): Component<T> => ({
   version: "placeholder", // Placeholder until we deprecate version in component definitions
   ...definition,
   documentationUrl: definition.documentationUrl || null,
   actions: Object.fromEntries(
-    Object.entries(definition.actions).map(([actionKey, action]) => [
+    Object.entries(definition?.actions || {}).map(([actionKey, action]) => [
```

Comment:
nit: I think `definition` is always defined so you don't need the optional chaining here.

Reply:
Thanks.